### ISSUE TITLE
Drop support PHP 7.4 and/or 8.0 ?

### ISSUE THREAD

**Issue by the-djmaze** [the-djmaze/snappymail#1453]
7.4 security support ended 28 Nov 2022
8.0 security support ended 26 Nov 2023

By only supporting PHP 8.1+, i can change the code to use native [enumerations](https://www.php.net/enumerations)

By only supporting PHP 8.0+, i can change the code to use [match](https://www.php.net/match) expressions.

This allows for better code handling and be less prone to bugs.

Are there still people using the unsupported PHP versions?

**Comment by phsc84** [comment]
+1 for dropping support of PHP 7.4 and 8.0

**Comment by mschilli87** [comment]
I am running SnappyMail on a Debian system with PHP 7.4.33. I would be willing to migrate but need some time to do so as this is system in production. If you let me know enough in advance, I'll look into it.

**Comment by the-djmaze** [comment]
> If you let me know enough in advance,

I will eventually :wink: It is just a thought at the moment.
Even I still have a server using PHP 7.4 due to some shopping software.

For now i could make a PHP 8.1+ branch (as the changes are minor) and see if both can be maintained for a while.

**Comment by the-djmaze** [comment]
Debian 12 has PHP 8.2
The issue is Debian 10 & 11 which need an additional repo like https://sury.org/

**Comment by Massedil** [comment]
I still use PHP 7.4 from Debian 11 bullseye.

**Comment by randolf** [comment]
> Debian 12 has PHP 8.2 The issue is Debian 10 & 11 which need an additional repo like https://sury.org/

That's true, although with OS upgrades PHP doesn't always get updated to the newest version, probably because there are some applications that break with PHP updates?  (I've seen this mostly with custom code at a number of sites -- we've gotten most of them to prioritize development to resolve this, but a few do lag behind for various reasons.)

Perhaps with such a significant change in requirements it may be helpful to release SnappyMail v3 with those requirements, and then gradually stop updating v2 (after updating v2 and v3 in parallel at least for a little while)?  This might help ease the transition for those who are still using older versions of PHP.  What do you think of this approach?

**Comment by the-djmaze** [comment]
> Perhaps with such a significant change in requirements it may be helpful to release SnappyMail v3 with those requirements.

That would be the best. Then when someone reports a bug, we at least know which code is used.

**Comment by apezio** [comment]
I am using PHP 7.4 on CentOS 7.  I have not found anything decent to replace it with - planning to keep running CentOS 7 for many more years.
PHP 7.4.33 (cli) (built: Aug  1 2023 09:00:17) ( NTS )
CentOS Linux release 7.9.2009 (Core)

**Comment by the-djmaze** [comment]
CentOS has the Remi repository.
I have no idea if Remi still supports 7
